ivadomed/utils.py

```diff
@@ -218,7 +218,11 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
     # load hard segmentation and label it
     nib_hard = nib.load(fname_hard)
     data_hard = nib_hard.get_fdata()
-    bin_struct = generate_binary_structure(3, 2)  # 18-connectivity
+    data_hard_l = []
+    for i in range(data_hard.shape[3]):
+        bin_struct = generate_binary_structure(3, 2)  # 18-connectivity
+        data_hard_tmp, n_l = label(data_hard[:, :, :, i], structure=np.array(bin_struct))
```

Comment:
this will break with 2d images. how is ivadomed project compatible with 2d input so far? it would be good to keep this in mind for new code, anticipating microscopy data in the near future

Reply:
To do--> change code so that it always consider the last dimension (wether it is 3D or 4D)

Reply:
done in bfa4d53. Changed something for swapping axis with condition (dim=3 or dim=4), a bit unsure about that (done in 946dd6a)